src/main/java/me/clip/placeholderapi/updatechecker/UpdateChecker.java

```diff
@@ -90,17 +91,22 @@ private boolean spigotIsNewer() {
       return false;
     }
 
-    String plV = toReadable(pluginVersion);
-    String spV = toReadable(spigotVersion);
-    return plV.compareTo(spV) < 0;
+    int[] plV = toReadable(pluginVersion);
+    int[] spV = toReadable(spigotVersion);
+
+    if (plV[0] < spV[0]) {
+      return true;
+    } else if ((plV[1] < spV[1])) {
+      return true;
+    } else return plV[2] < spV[2];
   }
 
-  private String toReadable(String version) {
-    if (version.contains("-DEV-")) {
-      version = version.split("-DEV-")[0];
+  private int[] toReadable(String version) {
+    if (version.contains("-DEV")) {
+      version = version.split("-DEV")[0];
     }
 
-    return version.replaceAll("\\.", "");
+    return Arrays.stream(version.split("\\.")).mapToInt(Integer::parseInt).toArray();
```

Comment:
And what if this will for whatever reason be not a Number?
This would throw an exception without it being handled at all.

Reply:
I doubt anyone will ever set the PAPI version number to a non-SemVer version. If they do, this error should be thrown, I think.